tests/test_chiral_restraints.py

```diff
@@ -626,5 +636,64 @@ def assert_same_signs(a, b):
         frame_bond_vols = get_chiral_bond_volumes(f)
         # f = f - np.mean(f, axis=0)
         # writer.write_frame(f*10)
-        assert_same_signs(frame_atom_vols, ref_chiral_atom_vols)
-        assert_same_signs(frame_bond_vols, ref_chiral_bond_vols)
+        if check_chiral_atoms:
+            assert_same_signs(frame_atom_vols, ref_chiral_atom_vols)
+        if check_chiral_bonds:
+            assert_same_signs(frame_bond_vols, ref_chiral_bond_vols)
+
+
+def test_chiral_inversion_in_single_topology():
+    very_short_hrex_params = replace(
+        DEFAULT_HREX_PARAMS, n_frames=2, n_eq_steps=2, steps_per_frame=2, hrex_params=HREXParams(n_frames_bisection=2)
+    )
+    ff = Forcefield.load_default()
+
+    # smi_a simplified from
+    # https://github.com/rdkit/rdkit/blob/06b135b4d54ea0c34d8a488df730390267bf5a36/Code/GraphMol/testChirality.cpp#L3213C18-L3213C41
+    smi_a = "C[C@]1(F)C[C@]([H])(O)C1"
+    smi_b = "C[C@]1(F)C[C@@]([H])(O)C1"  # invert one center, relative to mol_a
+    mol_a = Chem.AddHs(Chem.MolFromSmiles(smi_a))
+    mol_b = Chem.AddHs(Chem.MolFromSmiles(smi_b))
+    mol_a.SetProp("_Name", "A")
+    mol_b.SetProp("_Name", "B")
+    AllChem.EmbedMolecule(mol_a, randomSeed=2024)
+
+    def generate_more_complete_mapping():
+        """map whole ring + one substituent"""
+        substruct = Chem.MolFromSmiles("C1CCC1C")
+        atom_map = dict(zip(mol_a.GetSubstructMatch(substruct), mol_b.GetSubstructMatch(substruct)))
+        conf_a = mol_a.GetConformer()
+        coord_map = {atom_map[idx]: conf_a.GetAtomPosition(idx) for idx in atom_map}
+        AllChem.EmbedMolecule(
+            mol_b,
+            randomSeed=2024,
+            coordMap=coord_map,
+            useRandomCoords=True,
+            clearConfs=True,
+            enforceChirality=True,
+            useBasicKnowledge=True,
+        )
+        core = get_cores(mol_a, mol_b, **DEFAULT_ATOM_MAPPING_KWARGS)[0]
+        assert len(core) == 15
+        return mol_a, mol_b, core
+
+    def generate_less_complete_mapping():
+        """map part of ring + other substituent"""
+        substruct = Chem.MolFromSmiles("C1CCC1O")
+        atom_map = dict(zip(mol_a.GetSubstructMatch(substruct), mol_b.GetSubstructMatch(substruct)))
+        conf_a = mol_a.GetConformer()
+        coord_map = {atom_map[idx]: conf_a.GetAtomPosition(idx) for idx in atom_map}
+        AllChem.EmbedMolecule(mol_b, randomSeed=2045, coordMap=coord_map)
+        core = get_cores(mol_a, mol_b, **DEFAULT_ATOM_MAPPING_KWARGS)[0]
+        assert len(core) == 10
+
+        return mol_a, mol_b, core
+
+    # check that it runs
+    mol_a, mol_b, core = generate_more_complete_mapping()
+    _ = run_vacuum(mol_a, mol_b, core, ff, None, very_short_hrex_params, n_windows=3)
+
+    mol_a, mol_b, core = generate_less_complete_mapping()
+    _ = run_vacuum(mol_a, mol_b, core, ff, None, very_short_hrex_params, n_windows=3)
+
+    # TODO: slow, assert endstate chirality is preserved
```

Comment:
Would be good to check that the assertion fails without the use of chiral restraints (and then subsequently, fixed with the use of chiral restraints).